source/common/network/filter_manager_impl.h

```diff
@@ -17,15 +17,20 @@ class BufferSource {
 public:
   virtual ~BufferSource() {}
 
+  struct StreamBuffer {
+    Buffer::Instance& buffer;
+    bool last_byte;
```

Comment:
Do we get last_byte only on the read with the FIN, or on any subsequent reads?

Here's my use case.  We do onReadReady, read 60k and a FIN.  We end up backed up.  Upstream drains, and we readDisable(false) on the connection, so schedule a resumption event to kick off the pipeline.  I believe that results in on onReadReady() and doRead on a read-closed socket.

Is it the responsibility of the connection to remember it has received a FIN and not call onRead(), or is it the responsibility of the transport socket to remember it has received a FIN, skip the socket read, and return the FIN again?

 I think we don't currently handle this case, and I think it's worth latching the FIN in raw_buffer_socket, enforcing no future reads, and checking FIN status in connection_impl.cc (and anywhere else that does reads).  We should also comment this up in various places and add some regression tests if it's a real bug in the current impl.